### PR TITLE
Improvements for low bandwidth connections [DEVINFRA-616]

### PR DESCRIPTION
**Pull request by samvrlewis**
- Changes retry to terminate after 5 attempts (have made the number of attempts configurable too), rather than after 45 seconds
- Lowers the default tasks amounts
- Cleanup of some now unused config options

### CODE REVIEW

**Review by silverjam — APPROVED** [review]

**Comment by silverjam** [comment]
How does this impact performance for high bandwidth connections?

**Review by notoriaga — APPROVED** [review]
I'm curious how it compares to the aws cli too. Apparently they default to allowing 10 requests to aws in flight https://docs.aws.amazon.com/cli/latest/topic/s3-config.html#max-concurrent-requests

With these defaults we'd be at 32 right?

File: crates/esthri/src/ops/sync.rs [hunk 1]
```diff
@@ -177,6 +178,9 @@ fn create_dirent_stream<'a>(
                 yield Err(entry.err().unwrap().into());
                 return;
             };
+            if entry.file_name().to_string_lossy().contains(TEMP_FILE_PREFIX) {
```
Comment (samvrlewis) [inline crates/esthri/src/ops/sync.rs:181]: A somewhat unrelated fix, but have included it here just so we don't have branches on top of branches on top of branches. A little more info in the commit message: https://github.com/swift-nav/esthri/pull/255/commits/9cdafb7629c30edf547f2ce223b21258793b170c

**Comment by samvrlewis** [comment]
Some benchmarking of this here: https://swift-nav.atlassian.net/wiki/spaces/~460004640/pages/2137130948/Esthri+Benchmarking

Testing on a pod in an AWS environment (with high speed S3 transfers) esthri now is faster and either uses less memory or close to the amount of memory of the `aws tool`. Some sample results for all the non-Swift esthri devs out there. :grin: 

image

image

image

**Pull request merged**